src/controllers/auth.controllers.js

```diff
@@ -1,34 +1,43 @@
-import axios from "axios";
+import { JWT } from "google-auth-library";
 import dotenv from "dotenv";
+import serviceAccount from "../constants/serviceAccount.js";
 
 dotenv.config();
 
-//Get a TOKEN
-export const getAuthToken = async (req, res) => {
-  try {
-    const response = await axios.post(
-      process.env.TOKEN_URI,
-      {
-        client_id: process.env.CLIENT_ID,
-        client_secret: process.env.CLIENT_SECRET,
-        audience: process.env.AUTH0_AUDIENCE,
-        grant_type: "client_credentials",
-      },
-      {
-        headers: {
-          "Content-Type": "application/json",
-        },
-      },
-    );
+// Create a client with the service account
+const client = new JWT({
+  email: serviceAccount.client_email,
+  key: serviceAccount.private_key.replace(/\\n/g, "\n"),
+  scopes: ["https://www.googleapis.com/auth/cloud-platform"], //If we want to restrict the scope of the token, for example read&write only. This one is unrestricted.
+});
 
-    const token = response.data.access_token;
+// Function to get or renew a token
+const getAuthToken = async (req, res) => {
+  try {
+    await renewTokenIfNeeded();
+    const token = client.credentials.access_token;
 
     res.status(200).json({ token });
```

Comment:
let's follow the response convention

Reply:
We are returning the token so it will be stored on the frontend, Should it be different?

Reply:
You could do `({ data: token, error: null })`